src/map/pc.cpp

```diff
@@ -3246,6 +3246,9 @@ void pc_delautobonus(struct map_session_data &sd, std::vector<std::shared_ptr<s_
 				// Not all required items equipped anymore
 				restore = false;
 			}
+		} else {
+			// Not all required items equipped anymore
+			restore = false;
```

Comment:
this will make all autobonus after first `false` to fail
```suggestion
			it = bonus.erase(it);
			continue;
```